src/datadog_directive.h

```diff
@@ -7,47 +7,76 @@ extern "C" {
 #include <ngx_http.h>
 }
 
+#include <string_view>
+
+#include "string_util.h"
+
 namespace datadog {
 namespace nginx {
 
-char *delegate_to_datadog_directive_with_warning(ngx_conf_t *cf,
-                                                 ngx_command_t *command,
-                                                 void *conf) noexcept;
+/// Represent an NGINX configuration directive.
+/// This struct is there to leverage compile time generation of the module
+/// directives. because `ngx_command_t` is not constexpr-friendly.
+struct directive {
+  /// Function pointer on the function that will be called when the directive
+  /// will be processed.
+  using setter_func = auto(*)(ngx_conf_t *cf, ngx_command_t *cmd, void *conf)
+                          -> char *;
 
-char *add_datadog_tag(ngx_conf_t *cf, ngx_array_t *tags, ngx_str_t key,
-                      ngx_str_t value) noexcept;
+  /// Name of the directive.
+  std::string_view name;
+  /// Type of directive.
+  /// This value is a bitmask indicating the directive's applicable
+  /// configuration contexts * (e.g., `NGX_HTTP_MAIN_CONF`, `NGX_HTTP_SRV_CONF`,
+  /// etc.).
+  ngx_uint_t type;
+  /// Function pointer to the directive handler. This function is called when
+  /// the directive is encountered during configuration parsing.
+  setter_func callback;
+  /// Configuration flag for the directive.
+  ngx_uint_t conf;
+  /// Offset for storing the directive's value in a configuration structure.
+  ngx_uint_t offset;
+  /// Additional post-processing data.
+  void *post;
 
-char *set_datadog_tag(ngx_conf_t *cf, ngx_command_t *command,
-                      void *conf) noexcept;
+  ngx_command_t to_ngx_command() const {
+    return ngx_command_t{to_ngx_str(name), type, callback, conf, offset, post};
+  }
+};
 
-char *json_config_deprecated(ngx_conf_t *cf, ngx_command_t *command,
-                             void *conf) noexcept;
+template <std::size_t N1, std::size_t... Ns>
+constexpr auto merge_directives(const directive (&arr1)[N1],
```

Comment:
I trust you on this function but maybe it could be a good idea to add some comments as it is a bit an obscure function (from my point of view at least).

Reply:
Well, it's metaprogramming. If the reader has no knowledge of metaprogramming, it's sort of expected it seems a bit obscure.